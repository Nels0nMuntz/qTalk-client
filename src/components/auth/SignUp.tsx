import React from 'react';
import Link from 'next/link';
import { Icons } from '../Icons';
import SignUpForm from "./SignUpForm";
import SignInWithGoogleButton from "./SignInWithGoogleButton";
import { Separator } from "../ui/separator";

export default function SignUp() {
  return (
    <div className="container w-full sw:w-[400px] mx-auto flex flex-col justify-center space-y-6">
      <div className="w-full max-w-xs mx-auto flex flex-col space-y-4 text-center">
        <Icons.logo className="h-6 w-6 mx-auto" />
        <h1 className="text-2xl font-semibold tracking-tight">Welcome back</h1>
        <p className="text-sm mx-auto mb-4">
          By continuing, you are setting up a qTalk account and agree to our
          User Agreement and Privacy Policy
        </p>

        <SignInWithGoogleButton/>

        <div className="relative w-full text-center">
          <span className="relative inline-block px-3 text-sm text-zinc-500 uppercase bg-white z-[1]">or continue with</span>
          <Separator className="absolute top-1/2 left-0 right-0" />
        </div>

        <SignUpForm/>

        <p className="px-8 pt-4 text-center text-sm text-zinc-700">
          Already have an account?{' '}
          <Link
            href="/sign-in"
            className="text-sm underline underline-offset-4 hover:text-zinc-800"
          >
            Sign In
          </Link>
        </p>
      </div>
    </div>
  );
}
