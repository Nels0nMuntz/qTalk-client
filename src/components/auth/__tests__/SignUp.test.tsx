import { render } from '@testing-library/react';
import SignUp from '../SignUp';

const mockPush = jest.fn()
const mockRefresh = jest.fn()
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: mockPush,
    refresh: mockRefresh,
  }),
  useSearchParams: () => ({
    get: jest.fn(() => '/'),
  }),
}));

describe('SignUp component', () => {
  it('renders correctly', async () => {
    const { container } = render(<SignUp/>);
    expect(container).toMatchSnapshot();
  });
})