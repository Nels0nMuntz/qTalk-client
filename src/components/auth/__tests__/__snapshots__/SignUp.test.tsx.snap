// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component renders correctly 1`] = `
<div>
  <div
    class="container w-full sw:w-[400px] mx-auto flex flex-col justify-center space-y-6"
  >
    <div
      class="w-full max-w-xs mx-auto flex flex-col space-y-4 text-center"
    >
      <svg
        class="h-6 w-6 mx-auto"
        fill="#000000"
        preserveAspectRatio="xMidYMid meet"
        version="1.1"
        viewBox="0 0 36 36"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <title>
          talk-bubbles-solid
        </title>
        <path
          class="clr-i-solid clr-i-solid-path-1"
          d="M8,19V11H5a3,3,0,0,0-3,3V32a1,1,0,0,0,.56.89,1,1,0,0,0,1-.1L8.71,29H22.15A2.77,2.77,0,0,0,25,26.13V25H14A6,6,0,0,1,8,19Z"
        />
        <path
          class="clr-i-solid clr-i-solid-path-2"
          d="M31,4H14a3,3,0,0,0-3,3V19a3,3,0,0,0,3,3H27.55l4.78,3.71a1,1,0,0,0,1,.11,1,1,0,0,0,.57-.9V7A3,3,0,0,0,31,4Z"
        />
        <rect
          fill-opacity="0"
          height="36"
          width="36"
          x="0"
          y="0"
        />
      </svg>
      <h1
        class="text-2xl font-semibold tracking-tight"
      >
        Welcome back
      </h1>
      <p
        class="text-sm mx-auto mb-4"
      >
        By continuing, you are setting up a qTalk account and agree to our User Agreement and Privacy Policy
      </p>
      <button
        class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3 w-full"
        type="button"
      >
        <svg
          class="w-4 h-4 mr-2"
          data-testid="google-icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M22.56 12.25c0-.78-.07-1.53-.2-2.25H12v4.26h5.92c-.26 1.37-1.04 2.53-2.21 3.31v2.77h3.57c2.08-1.92 3.28-4.74 3.28-8.09z"
            fill="#4285F4"
          />
          <path
            d="M12 23c2.97 0 5.46-.98 7.28-2.66l-3.57-2.77c-.98.66-2.23 1.06-3.71 1.06-2.86 0-5.29-1.93-6.16-4.53H2.18v2.84C3.99 20.53 7.7 23 12 23z"
            fill="#34A853"
          />
          <path
            d="M5.84 14.09c-.22-.66-.35-1.36-.35-2.09s.13-1.43.35-2.09V7.07H2.18C1.43 8.55 1 10.22 1 12s.43 3.45 1.18 4.93l2.85-2.22.81-.62z"
            fill="#FBBC05"
          />
          <path
            d="M12 5.38c1.62 0 3.06.56 4.21 1.64l3.15-3.15C17.45 2.09 14.97 1 12 1 7.7 1 3.99 3.47 2.18 7.07l3.66 2.84c.87-2.6 3.3-4.53 6.16-4.53z"
            fill="#EA4335"
          />
          <path
            d="M1 1h22v22H1z"
            fill="none"
          />
        </svg>
        Continue with Google
      </button>
      <div
        class="relative w-full text-center"
      >
        <span
          class="relative inline-block px-3 text-sm text-zinc-500 uppercase bg-slate-50 z-[1]"
        >
          or continue with
        </span>
        <div
          class="shrink-0 bg-border h-[1px] w-full absolute top-1/2 left-0 right-0"
          data-orientation="horizontal"
          role="none"
        />
      </div>
      <form
        aria-label="signup form"
        class="space-y-4 text-left"
      >
        <div
          class="space-y-1"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for=":r0:-form-item"
          >
            Username
          </label>
          <input
            aria-describedby=":r0:-form-item-description"
            aria-invalid="false"
            class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
            id=":r0:-form-item"
            name="username"
            placeholder="your name"
            value=""
          />
        </div>
        <div
          class="space-y-1"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for=":r1:-form-item"
          >
            Email
          </label>
          <input
            aria-describedby=":r1:-form-item-description"
            aria-invalid="false"
            class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
            id=":r1:-form-item"
            name="email"
            placeholder="your@email.com"
            value=""
          />
        </div>
        <div
          class="space-y-1"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for=":r2:-form-item"
          >
            Password
          </label>
          <input
            aria-describedby=":r2:-form-item-description"
            aria-invalid="false"
            class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
            id=":r2:-form-item"
            name="password"
            placeholder="at least 6 characters"
            value=""
          />
        </div>
        <div
          class="space-y-1"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for=":r3:-form-item"
          >
            Confirm Password
          </label>
          <input
            aria-describedby=":r3:-form-item-description"
            aria-invalid="false"
            class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
            id=":r3:-form-item"
            name="confirmPassword"
            placeholder="the same as above"
            value=""
          />
        </div>
        <button
          class="inline-flex items-center justify-center text-sm font-medium ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-10 px-4 py-2 w-full text-white bg-rose-400 rounded-lg hover:bg-rose-500 transition-colors"
        >
          Get Started
        </button>
      </form>
      <p
        class="px-8 pt-4 text-center text-sm text-zinc-700"
      >
        Already have an account?
         
        <a
          class="text-sm underline underline-offset-4 hover:text-zinc-800"
          href="/sign-in"
        >
          Sign In
        </a>
      </p>
    </div>
  </div>
</div>
`;
